R/JoinKeys.R

```diff
@@ -370,6 +369,75 @@ join_keys <- function(...) {
   res
 }
 
+#' @title Getter for JoinKeys that returns the relationship between pairs of datasets
+#' @export
+`[.JoinKeys` <- function(x, dataset_1, dataset_2 = NULL) {
+  checkmate::assert_string(dataset_1)
+  checkmate::assert_string(dataset_2, null.ok = TRUE)
+
+  dataset_2 <- dataset_2 %||% dataset_1
+  x$get(dataset_1, dataset_2)
+}
+
+#' @rdname sub-.JoinKeys
+#' @export
+`[<-.JoinKeys` <- function(x, dataset_1, dataset_2 = NULL, value) {
+  checkmate::assert_string(dataset_1)
+  checkmate::assert_string(dataset_2, null.ok = TRUE)
+
+  dataset_2 <- dataset_2 %||% dataset_1
+  x$mutate(dataset_1, dataset_2, value)
+  x
+}
+
+#' @rdname join_keys
+#' @details
+#' `cdisc_join_keys` treat non-`JoinKeySet` arguments as possible CDISC datasets.
+#' The `dataname` is extrapolated from the name  (or fallback to the value itself if
+#' it's a `character(1)`).
+#'
+#' @export
+#' @examples
+#' cdisc_join_keys(join_key("dataset_A", "dataset_B", c("col_1" = "col_a")), "ADTTE")
+#'
+cdisc_join_keys <- function(...) {
```

Comment:
I just realized that we need `cdisc_join_keys` also in `ddl` constructor. It means that we won't have a named list of object - we can only have names of datasets.

```
ddl <- function(code, ui, server, input_mask, datanames, join_keys = cdisc_join_keys(datanames))

```

It means we need a cdisc_join_keys function which can just accept names of datasets. When I look at the function body, `name` (datanames) is exactly what this function needs. So I think that maybe we should have `datanames` argument instead of `...`? And do following

```r
cdisc_data(..., join_keys = cdisc_join_keys(names(...))
```

In above, in case of old-classes list will be unnamed so function will return NULL.

Reply:
So the function actually takes strings in `...` and parses them as if they are named lists.. the following calls are valid (as of now)

```R
# Identical output for all 3 calls
cdisc_join_keys(ADSL = rADSL, ADTTE = rADTTE)
cdisc_join_keys("ADSL", ADTTE = rADTTE)
cdisc_join_keys("ADSL", "ADTTE")
```

But we can also use your suggestion: `cdisc_data(..., join_keys = cdisc_join_keys(names(...))`

WDYT?